internal/characters/mika/attack.go

```diff
@@ -0,0 +1,93 @@
+package mika
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attacks"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/geometry"
+)
+
+const normalHitNum = 5
+
+// based on raiden frames
+// TODO: update frames, hitlags & hitboxes
+var (
+	attackFrames   [][]int
+	attackHitmarks = [][]int{{14}, {9}, {14}, {14, 27}, {34}}
+	// same between polearm and burst attacks so just use these arrays for both
+	attackHitlagHaltFrame = [][]float64{{0.02}, {0.02}, {0.02}, {0, 0}, {0.04}}
+	attackDefHalt         = [][]bool{{false}, {true}, {false}, {true, true}, {true}}
+	attackHitboxes        = [][]float64{{1.8, 2.8}, {2.5}, {1.8, 4}, {2.8}, {3}}
+	attackOffsets         = [][][]float64{{{0, -0.2}}, {{0, 0.5}}, {{0, -1.2}}, {{-0.5, 0.7}, {0.3, 0.7}}, {{0, 0}}}
+	attackFanAngles       = []float64{360, 270, 360, 360, 360}
```

Comment:
updated based on dm. The `Attack` logic will need to be changed to use circle vs box when relevant (remove raiden burst comment). Notes on the NA attacks:
* attack 4 is a 2 hit attack with identical hitboxes for both
* attack 2 and 4 are circle hitboxes, the rest are boxes (boxes have width and height in `attackHitboxes`)
* only attack 4 has an angle (attack 2 is a 360)
```golang
attackHitlagHaltFrame = [][]float64{{0.02}, {0.02}, {0.02}, {0, 0}, {0.04}}
attackDefHalt         = [][]bool{{false}, {true}, {false}, {true, true}, {true}}
attackHitboxes        = [][][]float64{{{1.8, 2.5}}, {{1.6}}, {{2.5, 2.5}}, {{4}, {4}}, {{1.5, 5}}}
attackOffsets         = []float64{0, 0.2, 0.5, -0.3, 1.0}
attackFanAngles       = []float64{360, 360, 360, 45, 360}
```